apps/meteor/app/api/server/v1/chat.js

```diff
@@ -547,7 +551,7 @@ API.v1.addRoute(
 				msg: new RegExp(escapeRegExp(text), 'i'),
 			};
 
-			const threadQuery = { ...query, ...typeThread, rid, tcount: { $exists: true } };
+			const threadQuery = { ...query, ...typeThread, rid: room._id, tcount: { $exists: true } };
```

Comment:
I know is OOS, but, would be good to move this to the model 😬

Reply:
Yeah, I agree, not a good practice to build a query on the route itself. However since this will be treated as a security fix, I'd prefer to keep the changes as minimal here and pick this up on Eng day in a separate PR